src/SimpleGraphs/generators/randgraphs.jl

```diff
@@ -1494,24 +1494,32 @@ function dorogovtsev_mendes(
     n < 3 && throw(DomainError("n=$n must be at least 3"))
     rng = rng_from_rng_or_seed(rng, seed)
     g = cycle_graph(3)
-
-    for iteration in 1:(n - 3)
-        chosenedge = rand(rng, 1:(2 * ne(g))) # undirected so each edge is listed twice in adjlist
-        u, v = -1, -1
-        for i in 1:nv(g)
-            edgelist = outneighbors(g, i)
-            if chosenedge > length(edgelist)
-                chosenedge -= length(edgelist)
-            else
-                u = i
-                v = edgelist[chosenedge]
-                break
-            end
-        end
-
-        add_vertex!(g)
+    bag_of_edges = Vector{SimpleEdge{Int}}(undef, 2 * n - 3) # Caching edges as they are added to avoid costly lookups
+
+    bag_of_edges[1] = SimpleEdge(1, 2)
+    bag_of_edges[2] = SimpleEdge(1, 3)
+    bag_of_edges[3] = SimpleEdge(2, 3)
+    index = 3
+
+    for _ in 1:(n - 3)
+        # Choose random edge from bag
+        edge = bag_of_edges[rand(rng, 1:index)]
+        u, v = edge.src, edge.dst
+
+        # Add new vertex
+        add_vertex!(g) || throw(
+            DomainError(
+                "Failed to add vertex. One possible explanation is that type $(eltype(g)) cannot represent enough vertices",
+            ),
```

Comment:
why is this necessary?

Reply:
After adding this I realized that it is currently note possible to use this function to generate a graph where the node types are different than Int64. Making this possible would better fit in a dedicated issue I think (and of coherence this would need to be done with other similar generators).

Currently, only this function allows for user-defined node types : 

`function SimpleDiGraph{T}(
    nv::Integer,
    ne::Integer;
    rng::Union{Nothing,AbstractRNG}=nothing,
    seed::Union{Nothing,Integer}=nothing,
) where {T<:Integer}`

If in the future we allow this kind of freedom, this checks makes sense as it could happen that someone tries to generate a graph too large for its nodes to be represented with the given type (this probably won't happen with Int64 though).

So as of now it isn't necessary.